File: Queue Reversal - GFG/queue-reversal.cpp
//{ Driver Code Starts
//Initial Template for C++

#include<bits/stdc++.h>
using namespace std;

// } Driver Code Ends
//function Template for C++

// //Function to reverse the queue.
// class Solution
// {
//     public:
//     queue<int> rev(queue<int> q)
//     {
//         stack<int> st;

//         while (!q.empty()) {
//             st.push(q.front());
//             q.pop();
//         }

//         while (!st.empty()) {
//             q.push(st.top());
//             st.pop();
//         }

//         return q;
//     }
// };
class Solution
{
    public:
    queue<int> rev(queue<int> q)
    {
          helper(q);
     	return q;// add code here.
    }
    
     void helper(queue<int>& q)
    {
        // base case
        if (q.size() == 0)
            return ;
     //store the ele
        int x = q.front();
        q.pop();
    // recur
         helper(q);
        //  push the ele back
        q.push(x);
    }
    
};

//{ Driver Code Starts.
int main()
{
    int test;
    cin>>test; 
    while(test--)
    {
    queue<int> q; 
    int n, var; 
    cin>>n; 
    while(n--)
    {
        cin>>var; 
        q.push(var);
    }
    Solution ob;
    queue<int> a=ob.rev(q); 
    while(!a.empty())
    {
        cout<<a.front()<<" ";
        a.pop();
    }
    cout<<endl; 
    }
}
// } Driver Code Ends

File: Queue Reversal - GFG/queue-reversal.java
//{ Driver Code Starts
import java.util.*;
import java.io.*;
class Reversing{
    public static void main(String[] args)throws IOException{
        BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
        
        //Reading total number of testcases
        int t=Integer.parseInt(br.readLine());
        while(t-->0){
            
            //Reading total number of elements
            int n=Integer.parseInt(br.readLine());
            
            //Creating a Queue
            Queue<Integer> q=new LinkedList<>();
            
            //Reading all the elements in a string
            String s=br.readLine();
            
            //Spliting the string into different
            //string separated by space
            String[] a=s.split(" ");
            
            //adding all the elements to the Queue
            for(String b:a){
                int x=Integer.parseInt(b);
                q.add(x);
            }
            
            //Creating an object of class Geeks
            GfG g=new GfG();
            
            //calling rev method of class Geeks
            q=g.rev(q);
            
            //printing the elements of the queue
            while(!q.isEmpty()){
                int x=q.peek();
                q.poll();
                System.out.print(x+" ");
            }
            System.out.println();
        }
    }
}
// } Driver Code Ends


//User function Template for Java
/*Complete the function below*/
class GfG{
    //Function to reverse the queue.
    public Queue<Integer> rev(Queue<Integer> q){
         Stack<Integer> st =  new Stack<>();
        while(!q.isEmpty())
        st.push(q.remove());
        
        while(!st.isEmpty())
        q.add(st.pop());
        
        
        return q;
    }
}